frameworks/rooch-nursery/sources/multisign_account.move

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// Bitcoin multisign account module
+module rooch_nursery::multisign_account{
+
+    use std::vector;
+    use std::option;
+    use moveos_std::signer;
+    use moveos_std::object::{Self, Object};
+    use rooch_framework::address_mapping;
+    use rooch_framework::bitcoin_address::{Self, BitcoinAddress};
+
+
+    const ErrorInvalidThreshold: u64 = 1;
+    const ErrorMultiSignAccountNotFound: u64 = 2;
+    const ErrorInvalidParticipant: u64 = 3;
+    const ErrorParticipantMustHasBitcoinAddress: u64 = 4;
+    const ErrorParticipantAlreadyJoined: u64 = 5;
+    const ErrorInvalidPublicKey: u64 = 6;
+
+    struct MultiSignAccountStore has key {
+    }
+
+    struct MultiSignAccount has key {
+        /// The multisign account address
+        bitcoin_address: BitcoinAddress,
+        /// The multisign account rooch address
+        address: address,
+        /// The multisign account threshold
+        threshold: u64,
+        /// The public keys of the multisign account
+        public_keys: vector<vector<u8>>,
+        /// The multisign account participants
+        participants: vector<address>,
+    }
+
+    public(friend) fun genesis_init(){
+        let account_store_obj = object::new_named_object(MultiSignAccountStore{});
+        object::transfer_extend(account_store_obj, @rooch_nursery);
+    }
+
+    fun borrow_mut_account_store() : &mut Object<MultiSignAccountStore> {
+        let object_id = object::named_object_id<MultiSignAccountStore>();
+        object::borrow_mut_object_extend<MultiSignAccountStore>(object_id)
+    }
+
+    fun borrow_account_store() : &Object<MultiSignAccountStore> {
+        let object_id = object::named_object_id<MultiSignAccountStore>();
+        object::borrow_object<MultiSignAccountStore>(object_id)
+    }
+
+    public entry fun create_multi_sign_account(
+        public_keys: vector<vector<u8>>,
+        threshold: u64,
+    ){
+        assert!(vector::length(&public_keys) >= threshold, ErrorInvalidThreshold);
+        let bitcoin_address = bitcoin_address::derive_multi_sign_address(&public_keys, threshold);
+        let address = bitcoin_address::to_rooch_address(&bitcoin_address);
+        let account_obj = object::new(MultiSignAccount {
+            bitcoin_address,
+            address,
+            threshold,
+            public_keys,
+            participants: vector::empty(),
+        });
+        let account_obj_id = object::id(&account_obj);
+        object::transfer_extend(account_obj, address);
+        let account_store = borrow_mut_account_store();
+        object::add_field(account_store, address, account_obj_id);
+    }
+
+    public entry fun join(sender: &signer, multi_sign_address: address, public_key: vector<u8>){
```

Comment:
It's hard to read the entire block of code. It may need some descriptions of the function.

Reply:
The PoC draft is just to figure out the requirement for the Bitcoin multi-sign account support.